File: src/main/java/org/shark/example/controller/rabbit/RabbitMessageQueueController.java
package org.shark.example.controller.rabbit;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.rabbitmq.SendRabbitMessageQueueService;
import org.shark.example.service.rabbitmq.pojo.MessageQueueInputDto;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/rabbit-mq")
@Slf4j
public class RabbitMessageQueueController {
    private final SendRabbitMessageQueueService sendRabbitMessageQueueService;

    @PostMapping("/send")
    public ResponseDto<Void> send(@RequestBody MessageQueueInputDto messageQueueInputDto) {
        return sendRabbitMessageQueueService.start(messageQueueInputDto);
    }
}

File: src/main/kotlin/org/shark/example/controller/rabbit/RabbitMessageQueueController.kt
package org.shark.example.controller.rabbit

import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.rabbitmq.SendRabbitMessageQueueService
import org.shark.example.service.rabbitmq.pojo.MessageQueueInputDto
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/rabbit-mq")
class RabbitMessageQueueController(
    private val sendRabbitMessageQueueService: SendRabbitMessageQueueService
) {
    private val log = LoggerFactory.getLogger(RabbitMessageQueueController::class.java)

    @PostMapping("/send")
    fun send(@RequestBody messageQueueInputDto: MessageQueueInputDto): ResponseDto<Void> {
        return sendRabbitMessageQueueService.start(messageQueueInputDto)
    }
}
